src/resources/scripts/toCoco.py

```diff
@@ -0,0 +1,116 @@
+import pandas as pd
+import numpy
+from shapely.geometry import Point
+from shapely.affinity import scale,rotate
+import warnings
+import sys
+import math
+import json
+import ast
+
+# target_file = sys.argv[2]
+# csv = sys.argv[3]
+
+# the path to the CSV file
+paths = sys.argv[3:]
+# json (output) path #ignore for now
+save_json_path = sys.argv[2]
+
+
+def check_shape(row):
+    # only polygons are accepted
+    shape = row.shape_name
+    flag_shape = 1
+    if not(shape == "LineString" or shape == "Polygon" or shape == "Rectangle" or shape=="Circle" or shape=="Ellipse") :
+        warnings.warn('The shape %s is not supported !' %(shape))
+        flag_shape = 0
+    return flag_shape
+
+def image(row):
+    image = {}
+    # row.attributes is a string, we want to transform it in a dict
+    desired_dict = json.loads(row.attributes)
+    image["height"] = desired_dict["height"]
+    image["width"] = desired_dict["width"]
+    image["id"] = int(row.image_id)
+    image["file_name"] = row.filename
+    image["longitude"] = float(row.image_longitude) if not math.isnan(row.image_longitude) else None
+    image["latitude"] = float(row.image_latitude) if not math.isnan(row.image_latitude) else None
+    return image
+
+def category(row):
+    category = {}
+    category["id"] = row.label_id
+    category["name"] = row.label_name
+    return category
+
+def annotation(row):
+    annotation = {}
+    desired_array = ast.literal_eval(row.points)
+    if row.shape_name=="Circle":                                         # convert Circle to Polygon using shapely
+        x,y,r=desired_array
+        circlePolygon=Point(x,y).buffer(r)
+        desired_array=numpy.array(list(zip(*circlePolygon.exterior.coords.xy))).flatten().astype(float).tolist()
+    elif row.shape_name=="Ellipse":                                      # convert Ellipse to Polygon using shapely
+        m1x,m1y,mi1x,mi1y,m2x,m2y,mi2x,mi2y=desired_array
+        x=(m1x+mi1x+m2x+mi2x)/4
+        y=(m1y+mi1y+m2y+mi2y)/4
+        lm=math.sqrt((m1x-m2x)**2+(m1y-m2y)**2)/2
+        lmi=math.sqrt((mi1x-mi2x)**2+(mi1y-mi2y)**2)/2
+        angle=math.atan((m1y-m2y)/(m1x-m2x))
+        circlePolygon=Point(x,y).buffer(1)                              # create Circle and...
+        circlePolygon=scale(circlePolygon, lm, lmi)                     # scale it to an ellipse and...
+        circlePolygon = rotate(circlePolygon,angle,use_radians=True)    # rotate it.
+        desired_array=numpy.array(list(zip(*circlePolygon.exterior.coords.xy))).flatten().astype(float).tolist()
```

Comment:
I would prefer a few more spaces here for better readability but it's up to you :wink: